content/authoring/tutorials/create-first-kata.mdx

```diff
@@ -32,175 +32,192 @@ To create a new kata, you need to earn the ["Create Kata" privilege](/gamificati
-
-_Unless_ the focus of your kata is debugging or your kata involves some deep C++ compile-time metaprogramming where 50% of the challenge itself is to make the code compile, you most certainly do not want the solver to waste their precious time fixing the initial solution just to get it to compile. The initial solution should provide the solver with a dummy implementation that "works" out of the box (possibly with runtime errors) such that they can start replacing the dummy implementations with their own code straight away. **It is a kata issue if the initial solution fails to compile, especially if it introduces unnecessary overhead for the solver.**
-
-Note that similar principles apply to interpreted languages (e.g. JavaScript, Python, Ruby): the initial solution should not contain syntax errors and/or (top-level) reference errors which may prevent the solver from getting started with the task immediately.
+The initial solution is the code users are provided with when they start training on your kata. How you set up your initial solution code will depend heavily on the discipline you selected. Bug-fixing and refactoring disciplines require the inclusion of a partially or fully working solution respectively within this block. For fundamentals and algorithm disciplines you will likely only include skeleton code, such as an empty function, class or other code that serves as a starting point for a user's solution. In some cases you may just want to include a few helpful comments to help get the user started, but no actual code itself.
+
+Special care must be taken when providing code for the initial solution. You may have encountered a situation yourself where the provided initial solution was incorrect (e.g. code that wouldn't compile, function signatures that differed from the tests or description, etc.). Hunting down the required information in the description or sample tests is an exercise in frustration and, more importantly, a waste of time that should have been spent actually *solving* the kata.
+
+The _only time_ non-compiling code is acceptable is when it is an _intrinsic part if the challenge_, for example when the kata focuses explicitly on debugging or involves solving language-specific compile-time problems. In any other case, **it is considered a kata issue of the initial solution does not compile**. To avoid this, you must obviously supply some kind of placeholder code or "dummy" implementation which, depending on the chosen language, typically takes the form of a no-op or return statement. Note that this placeholder code _only needs to satisfy the compiler_, but may (and likely will) lead to run-time errors if not replaced by the user.
```

Comment:
```suggestion
The _only time_ non-compiling code is acceptable is when it is an _intrinsic part if the challenge_, for example when the kata focuses explicitly on debugging or involves solving language-specific compile-time problems. In any other case, **it is considered a kata issue if the initial solution does not compile**. To avoid this, you must obviously supply some kind of placeholder code or "dummy" implementation which, depending on the chosen language, typically takes the form of a no-op or return statement. Note that this placeholder code _only needs to satisfy the compiler_, but may (and likely will) lead to run-time errors if not replaced by the user.
```